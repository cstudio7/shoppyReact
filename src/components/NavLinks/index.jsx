import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default class NavLinks extends Component {
  displayCorrectSidebar = (role) => {
    if (role === 'admin') {
      return (
        <li>
          <Link to="/attendants">Attendants</Link>
        </li>
      );
    }
    return (
      <li>
        <Link to="/sales">Sales</Link>
      </li>
    );
  };

  render() {
    const { userRole } = this.props;
    return (
      <ul>
        <li>
          <Link to="/">Dashboard</Link>
        </li>
        <li>
          <Link to="/categories">Categories</Link>
        </li>
        <li>
          <Link to="/products">Products</Link>
        </li>
        {this.displayCorrectSidebar(userRole)}
        <li>
          <Link to="/login">Logout</Link>
        </li>
      </ul>
    );
  }
}

NavLinks.propTypes = {
  userRole: PropTypes.string.isRequired
};
