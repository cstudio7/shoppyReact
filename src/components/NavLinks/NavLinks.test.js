import React from 'react';
import { shallow } from 'enzyme';
import NavLinks from './index';

describe('Test the NavLinks Component', () => {
  it('should match snapshot', () => {
    const userRole = 'attendant';
    const wrapper = shallow(<NavLinks userRole={userRole} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot', () => {
    const userRole = 'admin';
    const wrapper = shallow(<NavLinks userRole={userRole} />);
    wrapper.instance().handleLogout();
    expect(wrapper).toMatchSnapshot();
  });
});
