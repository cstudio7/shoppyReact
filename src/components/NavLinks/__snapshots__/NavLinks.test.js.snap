// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the NavLinks Component should match snapshot 1`] = `
<ul>
  <li>
    <Link
      replace={false}
      to="/"
    >
      Dashboard
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/categories"
    >
      Categories
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/products"
    >
      Products
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/cart"
    >
      My Cart 
    </Link>
  </li>
  <li
    onClick={[Function]}
    role="presentation"
  >
    <Link
      replace={false}
      to="/login"
    >
      Logout
    </Link>
  </li>
</ul>
`;

exports[`Test the NavLinks Component should match snapshot 2`] = `
<ul>
  <li>
    <Link
      replace={false}
      to="/"
    >
      Dashboard
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/categories"
    >
      Categories
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/products"
    >
      Products
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/attendants"
    >
      Attendants
    </Link>
  </li>
  <li
    onClick={[Function]}
    role="presentation"
  >
    <Link
      replace={false}
      to="/login"
    >
      Logout
    </Link>
  </li>
</ul>
`;
